README.md

```diff
@@ -1,15 +1,226 @@
+The pipeline is written in Nextflow and uses Singularity/Docker to run containerised tools.
+
+There are two main steps to this workflow 
+- First step is to generate a large set of candidate somatic variants using the tool  [Mutect2](https://gatk.broadinstitute.org/hc/en-us/articles/360035531132). 
+- The next step is to filter the candidate variants to obtain a more confident set of somatic variant calls. 
+- The third (optional) step is to annotate the variants with gene level information using an external genomic variant annotations and functional effect prediction tool called [SnpEff](http://pcingola.github.io/SnpEff/).
+
+## Diagram
+
+<p align="center"> 
+<img src="./images/Somatic_variant_calling.png" width="60%">
+</p> 
+
 
 ## Workflow description 
 
 ## User guide 
 
-## Component tools 
+To run this pipeline, you will need to prepare your input files, reference data, and clone this repository. Before proceeding, ensure Nextflow is installed on the system you're working on. To install Nextflow, see these [instructions](https://www.nextflow.io/docs/latest/getstarted.html#installation). 
+
+### 1. Prepare inputs
+
+To run this pipeline you will need the following inputs: 
+
+* Paired Tumor-Normal paired BAM files
+* Corresponding BAM index files  
+* Input sample sheet 
+
+This pipeline processes paired BAM files and is capable of processing multiple samples in parallel. BAM files are expected to be coordinate sorted and indexed (see [Fastq-to-BAM](https://github.com/Sydney-Informatics-Hub/Fastq-to-BAM) for an example of a best practice workflow that can generate these files).  
+
+You will need to create a sample sheet with information about the samples you are processing, before running the pipeline. This file must be **comma-separated** and contain a header and one row per sample. Columns should correspond to sampleID, BAM-N file-path, BAM-T file-path: 
+
+```csv
+sampleID,bam-N,bam-T 
+SAMPLE1,/data/Bams/sample1-N.bam,/data/Bams/sample1-T.bam
+SAMPLE2,/data/Bams/sample2-N.bam,/data/Bams/sample2-T.bam|
+``````
+
+When you run the pipeline, you will use the mandatory `--input` parameter to specify the location and name of the input file: 
+
+```
+--input /path/to/samples.csv
+```
+
+### 2. Prepare the reference materials 
+
+To run this pipeline you will need the following reference files:
+
+* Indexed reference genome in FASTA format 
+
+You will need to download and index a copy of the reference genome you would like to use. Reference FASTA files must be accompanied by a .fai index file. If you are working with a species that has a public reference genome, you can download FASTA files from the [Ensembl](https://asia.ensembl.org/info/data/ftp/index.html), [UCSC](https://genome.ucsc.edu/goldenPath/help/ftp.html), or [NCBI](https://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq/) ftp sites. You can use our [IndexReferenceFasta-nf pipeline](https://github.com/Sydney-Informatics-Hub/IndexReferenceFasta-nf) to generate indexes. 
```

Comment:
Was this also taken directly from GSV? Does this actually apply to the Somatic pipeline or are there other important considerations? Please don't copy stuff over from other workflows without checking.

Reply:
Checked now.

GATK (the only tool) required a indexed reference fasta as input. So the above statement is valid here.

Reply:
GATK typically requires specific indexes. The indexes currently on gdata that you're using comprise bwa, samtools, and gatk. Ensure you know exactly which indexes need to be produced.

Reply:
thanks.

- **samtools**      .fai (for indexing FASTA files)
- **bwa**.    .amb, .ann, .bwt, .pac, .sa (for BWA index files)
- **picard** : .dict (for sequence dictionaries)

I will include these details about what and how they can be generated

Reply:
Are all of these ae actually needed to execute the processes though

Reply:
- We require  **.fai** 
- We start with bam - So bwa is used by [Fastq-to-BAM](https://github.com/Sydney-Informatics-Hub/Fastq-to-BAM) which we recommend in the Readme.md. So we may not want to mention it in this Readme...
-  **.dict** is used for estimating intervals in generating intervals files based on "input interval number" as shown [here](https://github.com/Sydney-Informatics-Hub/Fastq-to-BAM/blob/fastq-to-bam-v2/create_gatk_ref_intervals.pbs). However if we go by using the "input interval number" to get exact intervals, we do not require it. I will confirm.